File: src/main/java/com/linebot/repository/NoticeRepository.java
package com.linebot.repository;

import com.linebot.entity.Notice;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface NoticeRepository extends JpaRepository<Notice, Long> {
    List<Notice> findByTypeAndDeleted(int type, String deleted);
    List<Notice> findByUserIdAndDeleted(String userId, String deleted);
    List<Notice> findByUserIdAndType(String userId, int type);
}

File: src/main/kotlin/com/linebot/repository/NoticeRepository.kt
package com.linebot.repository

import com.linebot.entity.Notice
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface NoticeRepository : JpaRepository<Notice, Long> {
    fun findByTypeAndDeleted(type: Int, deleted: String): List<Notice>
    fun findByUserIdAndDeleted(userId: String, deleted: String): List<Notice>
    fun findByUserIdAndType(userId: String, type: Int): List<Notice>
}
